src/groups/mqb/mqbc/mqbc_storagemanager.cpp

```diff
@@ -3448,6 +3414,50 @@ void StorageManager::stop()
                              bdlf::PlaceHolders::_2));  // latch
 }
 
+void StorageManager::initializeQueueKeyInfoMap(const mqbc::ClusterState* clusterState)
+{
+    // executed by the *CLUSTER DISPATCHER* thread
+
+    // PRECONDITIONS
+    BSLS_ASSERT_SAFE(d_dispatcher_p->inDispatcherThread(d_cluster_p));
+    BSLS_ASSERT_SAFE(clusterState);
+
+    if (d_isQueueKeyInfoMapVecInitialized) {
+        // The queue key info map vec should only be initialized once.
```

Comment:
Does it make sense to log here?  Or do we want the semantics of this to be silently idempotent?  I see a comment later on that indicates this must be initialized only once, so it seems like it'd be good to know if we accidentally do it twice.

An `ASSERT` would feel heavy-handed, probably, but maybe some indication would be good.

Reply:
By FSM machinery, `initializeQueueKeyInfoMap()` is called as the last step of transitioning from [`Healing Follower` to `Healed Follower`](https://github.com/kaikulimu/blazingmq/blob/f3d03d74c4d6bb335080b1d4120d58bcaebe5fc3/src/groups/mqb/mqbc/mqbc_clusterstatetable.h#L402-L405), or from [`Healing Primary` to `Healed Primary`](https://github.com/kaikulimu/blazingmq/blob/f3d03d74c4d6bb335080b1d4120d58bcaebe5fc3/src/groups/mqb/mqbc/mqbc_clusterstatetable.h#L519-L522). Normally, this method should only be called once. However, if there are fault scenarios where a node can go back-and-forth between healing and healed, then this method can be called more than once. I guess in that uncommon scenario, it's still a good thing to add a log. Will add.

Reply:
d